api/client/build.go

```diff
@@ -296,6 +306,10 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		headers:     headers,
 	}
 
+	if *suppressOutput {
+		sopts.rawTerminal = false
+		sopts.out = bytes.NewBuffer(nil)
```

Comment:
Just wondering if using `outBuf` here would work? Then we'd only have one buffer to worry about later on.
And related to this, down on line 343 you print `sopts.out` but I didn't see you print `outBuf`. Did I miss it or is that an oversight?